### ISSUE TITLE
Docker 1.11.2 crash with http: panic serving @: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by resnostyle** [docker/docker#25663]
**Output of `docker version`:**

```
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64
```
**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 3
Server Version: 1.11.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 27
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 3.13.0-86-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.954 GiB
Name: ip-10-15-0-61
ID: 6KLZ:ZSZ4:D3UW:AC3A:7OGM:6FCV:WEJY:B5H7:MRIF:4B2G:PFNT:7TOF
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```
This is an AWS Ubuntu server.

```
2016/08/12 13:28:24 http: panic serving @: runtime error: invalid memory address or nil pointer dereference
goroutine 3454 [running]:
net/http.(*conn).serve.func1(0xc821039970, 0x7fb6af9331d0, 0xc820b44078)
	/usr/local/go/src/net/http/server.go:1287 +0xb5
net.networkNumberAndMask(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/ip.go:433 +0x42
net.(*IPNet).Contains(0x0, 0xc820cba490, 0x4, 0x4, 0xc820cc4800)
	/usr/local/go/src/net/ip.go:457 +0x25
github.com/docker/libnetwork/drivers/bridge.(*networkConfiguration).conflictsWithNetworks(0xc8210510e0, 0xc820de0880, 0x40, 0xc820b4f800, 0xb, 0xb, 0x0, 0x0)
	/usr/src/docker/vendor/src/github.com/docker/libnetwork/drivers/bridge/bridge.go:345 +0x40b
github.com/docker/libnetwork/drivers/bridge.(*driver).createNetwork(0xc8203e19a0, 0xc8210510e0, 0x0, 0x0)
	/usr/src/docker/vendor/src/github.com/docker/libnetwork/drivers/bridge/bridge.go:626 +0x80d
github.com/docker/libnetwork/drivers/bridge.(*driver).CreateNetwork(0xc8203e19a0, 0xc820de0880, 0x40, 0xc82126bfb0, 0xc820ff8f90, 0x1, 0x1, 0x2213f28, 0x0, 0x0, ...)
	/usr/src/docker/vendor/src/github.com/docker/libnetwork/drivers/bridge/bridge.go:573 +0x419
github.com/docker/libnetwork.(*controller).addNetwork(0xc82039c840, 0xc8213ccfc0, 0x0, 0x0)
	/usr/src/docker/vendor/src/github.com/docker/libnetwork/controller.go:512 +0x18e
github.com/docker/libnetwork.(*controller).NewNetwork(0xc82039c840, 0x173d898, 0x6, 0xc82126bce0, 0x27, 0xc820cccc40, 0x4, 0x4, 0x0, 0x0, ...)
	/usr/src/docker/vendor/src/github.com/docker/libnetwork/controller.go:471 +0x493
github.com/docker/docker/daemon.(*Daemon).CreateNetwork(0xc82032be00, 0xc82126bce0, 0x27, 0x173d898, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/src/docker/.gopath/src/github.com/docker/docker/daemon/network.go:114 +0x3e1
github.com/docker/docker/api/server/router/network.(*networkRouter).postNetworkCreate(0xc8207771a0, 0x7fb6af934d68, 0xc82126bbc0, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00, 0xc82126bb30, 0x0, 0x0)
	/usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/network/network_routes.go:94 +0x945
github.com/docker/docker/api/server/router/network.(*networkRouter).(github.com/docker/docker/api/server/router/network.postNetworkCreate)-fm(0x7fb6af934d68, 0xc82126bbc0, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00, 0xc82126bb30, 0x0, 0x0)
	/usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/network/network.go:31 +0x74
github.com/docker/docker/api/server/middleware.NewVersionMiddleware.func1.1(0x7fb6af934d68, 0xc82126bbc0, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00, 0xc82126bb30, 0x0, 0x0)
	/usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware/version.go:42 +0xa43
github.com/docker/docker/api/server/middleware.NewUserAgentMiddleware.func1.1(0x7fb6af934d68, 0xc82126bb90, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00, 0xc82126bb30, 0x0, 0x0)
	/usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware/user_agent.go:34 +0x6b8
github.com/docker/docker/api/server.(*Server).makeHTTPHandler.func1(0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00)
	/usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:132 +0x138
net/http.HandlerFunc.ServeHTTP(0xc82089b2e0, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00)
	/usr/local/go/src/net/http/server.go:1422 +0x3a
github.com/gorilla/mux.(*Router).ServeHTTP(0xc8201cf900, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00)
	/usr/src/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x29e
github.com/docker/docker/api/server.(*routerSwapper).ServeHTTP(0xc8208c9a50, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00)
	/usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router_swapper.go:29 +0x72
net/http.serverHandler.ServeHTTP(0xc820334cc0, 0x7fb6ac02de70, 0xc820db1e40, 0xc820cdae00)
	/usr/local/go/src/net/http/server.go:1862 +0x19e
net/http.(*conn).serve(0xc821039970)
	/usr/local/go/src/net/http/server.go:1361 +0xbee
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1910 +0x3f6
```

**Steps to reproduce the issue:**
1.  Under our CI environment we start up several containers with docker-compose. Each of those containers starts in a namespace container and network. 
2. While the node is pulling code and starting up containers docker will error out with the attached stacktrace.  Docker-compose will no longer be able to interact with docker and displays the error "Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?"
3. Until at which time I restart the docker daemon docker-compose will be unable to reconnect and issue commands. 

During our start-up we bring two nodes up, this particular node could possibly be downloading more data then the other and could be under heavier load.

**Comment by cpuguy83** [comment]
Thanks, looks like this is fixed already in 1.12

**Comment by lox** [comment]
@cpuguy83 any insight into what the cause and / or fix was? Seeing this issue with docker 1.13.1

**Comment by thaJeztah** [comment]
@lox may be best to open a new issue with more information (and the panic/logs) so that it can be looked into, and confirmed it's the same issue or a different one
